test/langtools/tools/lib/toolbox/ToolBox.java

```diff
@@ -180,9 +180,41 @@ public void checkEqual(List<String> l1, List<String> l2) throws Error {
      * @return the strings matching the regular expression
      */
     public List<String> grep(Pattern pattern, List<String> lines) {
-        return lines.stream()
-                .filter(s -> pattern.matcher(s).find())
-                .collect(Collectors.toList());
+        return grep(pattern, lines, false);
+    }
+
+    /**
+     * Filters a list of strings according to the given regular expression.
+     * @param regex the regular expression
+     * @param lines the strings to be filtered
+     * @param invert identify positive or negative filtering
+     *               true: negative filtering, return the unmatched strings
+     *               false: positive filtering, return the matched strings
```

Comment:
This is not a particularly well-formed javadoc comment.  Imagine the output if you were to run the file through javadoc.

Suggest:
```
Filters a list of strings according to the given regular expression,
returning either the strings that match or the strings that do not match.
@param regex the regular expression
@param lines the strings to be filtered
@param invert if true, return the lines that do not match; otherwise if false, return the lines that do match.
```

The `invert` parameter feels "inverted" leading to a "double negative".

Maybe it would be better to call the parameter "match" and invert the sense, so the doc comment reads:

```
Filters a list of strings according to the given regular expression,
returning either the strings that match or the strings that do not match.
@param regex the regular expression
@param lines the strings to be filtered
@param match if true, return the lines that match; otherwise if false, return the lines that do not match.
```

Reply:
Fixed.